tests/test_router_ospf_area.rb

```diff
@@ -188,6 +188,273 @@ def test_range
+    assert_equal(ad.default_nssa_enable, ad.nssa_enable)
+    ad.nssa_set(true, ad.default_nssa_def_info_originate,
+                ad.default_nssa_no_redistribution,
+                ad.default_nssa_no_summary,
+                ad.default_nssa_route_map)
+    assert_equal(true, ad.nssa_enable)
+    assert_equal(ad.default_nssa_def_info_originate, ad.nssa_def_info_originate)
+    assert_equal(ad.default_nssa_no_redistribution, ad.nssa_no_redistribution)
+    assert_equal(ad.default_nssa_no_summary, ad.nssa_no_summary)
+    assert_equal(ad.default_nssa_route_map, ad.nssa_route_map)
+
+    ad.nssa_set(ad.default_nssa_enable,
+                ad.default_nssa_def_info_originate,
+                ad.default_nssa_no_redistribution,
+                ad.default_nssa_no_summary,
+                ad.default_nssa_route_map)
+    assert_equal(ad.default_nssa_enable, ad.nssa_enable)
+    assert_equal(ad.default_nssa_def_info_originate, ad.nssa_def_info_originate)
+    assert_equal(ad.default_nssa_no_redistribution, ad.nssa_no_redistribution)
+    assert_equal(ad.default_nssa_no_summary, ad.nssa_no_summary)
+    assert_equal(ad.default_nssa_route_map, ad.nssa_route_map)
+
+    av = create_routerospfarea_vrf
+    assert_equal(av.default_nssa_enable, av.nssa_enable)
+    av.nssa_set(true, av.default_nssa_def_info_originate,
+                av.default_nssa_no_redistribution,
+                av.default_nssa_no_summary,
+                av.default_nssa_route_map)
+    assert_equal(true, av.nssa_enable)
+    assert_equal(av.default_nssa_def_info_originate, av.nssa_def_info_originate)
+    assert_equal(av.default_nssa_no_redistribution, av.nssa_no_redistribution)
+    assert_equal(av.default_nssa_no_summary, av.nssa_no_summary)
+    assert_equal(av.default_nssa_route_map, av.nssa_route_map)
+
+    av.nssa_set(av.default_nssa_enable,
+                av.default_nssa_def_info_originate,
+                av.default_nssa_no_redistribution,
+                av.default_nssa_no_summary,
+                av.default_nssa_route_map)
+    assert_equal(av.default_nssa_enable, av.nssa_enable)
+    assert_equal(av.default_nssa_def_info_originate, av.nssa_def_info_originate)
+    assert_equal(av.default_nssa_no_redistribution, av.nssa_no_redistribution)
+    assert_equal(av.default_nssa_no_summary, av.nssa_no_summary)
+    assert_equal(av.default_nssa_route_map, av.nssa_route_map)
+  end
+
+  def test_nssa_default_vrf_others
+    ad = create_routerospfarea_default
+    # on n8k (only), we cannot configure
+    # "area <area> nssa default-information-originate",
+    # properly if we reset it first. It is only configuring nssa
+    # but not the other parameters. bug ID: CSCva11482
+    ad.nssa_set(true, true,
+                ad.default_nssa_no_redistribution,
+                ad.default_nssa_no_summary,
+                ad.default_nssa_route_map)
+    assert_equal(true, ad.nssa_enable)
+    assert_equal(true, ad.nssa_def_info_originate) unless
+      /N8/ =~ node.product_id
```

Comment:
You shouldn't skip this for 8k - if/when they fix it we'll likely forget to update this test. Better to explicitly test for it so that it fails when they fix it; i.e.:

```
  if node.product_id[/N8/]
    refute(ad.nssa_def_info_originate)
  else
    assert(ad.nssa_def_info_originate)
  end
```